src/intro-to-c/debugging/general-debugging.md

```diff
@@ -0,0 +1,293 @@
+---
+title: General Debugging
+author: Johnson Laguerre
+---
+
+One of the classic debugging approaches is print-based debugging. Strategically-placed print statements let you see what is happening in your program at "critical" points in time and can help you pinpoint where your program is going wrong.
+
+Below, we will take a look at some examples of print-based debugging helping us locate errors.
+
+# Example A: Alarming Arrays.
```

Comment:
Maybe I am just missing something, but what are the print statements here used for? Is this for debugging or the actual functionality of the program itself? I ask this since it appears like this program would not have the stack smashing problem should there have been no print statements in the first place. (I would recommend showing the stack smashing error first in isolation, then adding the print statements seen in the first code example. In reality, the stack smashing check created by GCC would always occur at the conclusion of a function.)